### PR TITLE
 neovim-qt: wrap in separate derivation

### PR DESCRIPTION
**Pull request by timokau**
###### Motivation for this change

This makes it possible to swap out the (wrapped) neovim without
recompiling neovim-qt. In particular, the user can use `neovim.override`
to configure their neovim and then use that same configuration for
neovim-qt, without having to give up binary caching.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by timokau** [comment]
Someone should make sure I didn't mess up the darwin wrapper @NixOS/darwin-maintainers. You can test ist by building and executing

```nix
neovim-qt.override {
  neovim = neovim.override {
    configure = {
      customRC = ''
        normal iIt worked!
      '';
    };
  };
}
```

Which should then start neovim-qt and show the text "It worked!".

**Comment by timokau** [comment]
CC @peterhoeg

**Comment by timokau** [comment]
@GrahamcOfBorg build neovim-qt

File: pkgs/applications/editors/neovim/qt.nix
Comment (grahamc) [inline pkgs/applications/editors/neovim/qt.nix]: I'm not sure I like this change in general, but I'll leave that up to @peterhoeg.

Something which is important, though, is this new package has no meta -- the meta on the unwrapped one should likely be inherited here.

Comment (grahamc) [reply pkgs/applications/editors/neovim/qt.nix]: (and yeah, this is why ofborg didn't ping @peterhoeg)

Comment (timokau) [reply pkgs/applications/editors/neovim/qt.nix]: Right, fixed.

File: pkgs/applications/editors/neovim/qt.nix
Comment (bhipple) [inline pkgs/applications/editors/neovim/qt.nix]: You might be able to run this with:

```
export QT_PLUGIN_PATH ${qtbase}/${qtbase.qtPluginPrefix}
```
as a stop-gap solution until we get the global fix in https://github.com/NixOS/nixpkgs/pull/44047

Comment (timokau) [reply pkgs/applications/editors/neovim/qt.nix]: I tried that, it didn't fix the issue:

```
4/7 Test #7: tst_shell ........................***Failed    0.09 sec
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/build/runtime-nixbld'
qt.qpa.screen: QXcbConnection: Could not connect to display 
Could not connect to any X display.
```

Comment (bhipple) [reply pkgs/applications/editors/neovim/qt.nix]: Interesting, that suggests it's unlikely to work even with #44047, since it requires not just QT plugins but a full X display.

Comment (timokau) [reply pkgs/applications/editors/neovim/qt.nix]: You're right. I've amended that comment.

**Comment by timokau** [comment]
The new darwin failure is unrelated (caused by a cups failure).

**Comment by timokau** [comment]
Does anyone have objections to merging this? (@peterhoeg?)

**Review by peterhoeg — APPROVED** [review]

**Pull request merged**
